### PR TITLE
[TT-12589] Update documentation for master Pump 1.11

### PR DESCRIPTION
**Pull request by buger**
Triggered by: mativm02

Included:

Tyk Gateway: false
Tyk Dashboard: false
Tyk MDCB false
Tyk Pump true

Intended for: master
Changes sourced from: master
Config info generator branch: main

Note: Pump Config Generator (branch suffix: TT-12589)

JIRA: https://tyktech.atlassian.net/browse/TT-12589

### CODE REVIEW

File: tyk-docs/content/shared/pump-config.md [hunk 1]
```diff
@@ -1796,6 +1796,178 @@ Type: `string`<br />
...
+Type: `string`<br />
+
+The name of the pump. This is used to identify the pump in the logs.
+Deprecated, use `type` instead.
+
+### pumps.kinesis.type
+ENV: <b>TYK_PMP_PUMPS_KINESIS_TYPE</b><br />
+Type: `string`<br />
+
+Sets the pump type. This is needed when the pump key does not equal to the pump name type.
```
Comment (dcs3spp) [inline tyk-docs/content/shared/pump-config.md:1810]:
```suggestion
Sets the pump type. This is needed when the pump key does not equal the pump name type.
```

File: tyk-docs/content/shared/pump-config.md [hunk 2]
```diff
@@ -1796,6 +1796,178 @@ Type: `string`<br />
...
+Current valid types are: `mongo`, `mongo-pump-selective`, `mongo-pump-aggregate`, `csv`,
+`elasticsearch`, `influx`, `influx2`, `moesif`, `statsd`, `segment`, `graylog`, `splunk`, `hybrid`, `prometheus`,
+`logzio`, `dogstatsd`, `kafka`, `syslog`, `sql`, `sql_aggregate`, `stdout`, `timestream`, `mongo-graph`,
+`sql-graph`, `sql-graph-aggregate`, `resurfaceio`.
+
+### pumps.kinesis.filters
+This feature adds a new configuration field in each pump called filters and its structure is
+the following:
+```{.json}
```
Comment (dcs3spp) [inline tyk-docs/content/shared/pump-config.md:1819]: Replace this instance and remaining instances with 

```suggestion
```json
```

File: tyk-docs/content/shared/pump-config.md [hunk 3]
```diff
@@ -1796,6 +1796,178 @@ Type: `string`<br />
...
+"filters":{
+  "api_ids":[],
+  "org_ids":[],
+  "response_codes":[],
+  "skip_api_ids":[],
+  "skip_org_ids":[],
+  "skip_response_codes":[]
+}
+```
+The fields api_ids, org_ids and response_codes works as allow list (APIs and orgs where we
```
Comment (dcs3spp) [inline tyk-docs/content/shared/pump-config.md:1829]: Add backticks to the fields, something like:

The fields `api_ids`, `org_ids` and `response_codes` represent an allow list (APIs and orgs where we want to send the analytics records). The fields `skip_api_ids`, `skip_org_ids` and
`skip_response_codes` represent a block list.

File: tyk-docs/content/shared/pump-config.md [hunk 4]
```diff
@@ -1796,6 +1796,178 @@ Type: `string`<br />
...
+   "org_ids": ["org1","org2"]
+ },
+ "meta": {
+   "csv_dir": "./bar"
+ }
+}
+```
+
+### pumps.kinesis.filters.org_ids
+ENV: <b>TYK_PMP_PUMPS_KINESIS_FILTERS_ORGSIDS</b><br />
```
Comment (dcs3spp) [inline tyk-docs/content/shared/pump-config.md:1849]: should _ORGS_IDS and _ORGSIDS in env var names be _ORG_IDS and _ORG_IDS?

**Review by dcs3spp — CHANGES_REQUESTED** [review]
Thanks for submitting PR @JoanCamosTyk @mativm02 

PR LGTM! Made small grammar suggestions and some minor suggestions / question:
- replace ```{.json} with ```json
- should _ORGS_IDS and _ORGSIDS in env var names be _ORG_IDS and _ORG_IDS?

Can a peer review be arranged to check that all required new config fields are included in the PR?

File: tyk-docs/content/shared/pump-config.md (hunk 4 above)
Comment (mativm02) [reply tyk-docs/content/shared/pump-config.md:1849]: @dcs3spp the "_" separator indicates when a field is a child of a parent struct. In this case, ORGSIDS is a child of FILTERS struct

**Review by dcs3spp — APPROVED** [review]
Thanks @mativm02 PR LGTM I will merge and release when Pump 1.11 release has been announced

**Pull request merged**